packages/storybook/stories/va-number-input.stories.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { generateEventsDescription } from './events';
+import { getWebComponentDocs, propStructure } from './wc-helpers';
+
+const numberInputDocs = getWebComponentDocs('va-number-input');
+
+export default {
+  title: 'Components/va-number-input',
+  parameters: {
+    docs: {
+      description: {
+        component: generateEventsDescription(numberInputDocs),
+      },
+    },
+  },
+  argTypes: {
+    inputmode: {
+      control: {
+        type: 'select',
+        options: ['decimal', 'numeric'],
+      },
+    },
+  },
+};
+
+const defaultArgs = {
+  'name': 'my-input',
+  'label': 'My input',
+  'enable-analytics': false,
+  'required': false,
+  'error': undefined,
+  'value': 0,
+  'inputmode': 'numeric',
+  'aria-describedby': undefined,
+  'min': undefined,
+  'max': undefined,
+};
+
+const Template = ({
+  name,
+  label,
+  'enable-analytics': enableAnalytics,
+  required,
+  error,
+  value,
+  inputmode,
+  'aria-describedby': ariaDescribedby,
+  min,
+  max,
+}) => {
+  return (
+    <va-number-input
+      name={name}
+      label={label}
+      enable-analytics={enableAnalytics}
+      required={required}
+      error={error}
+      value={value}
+      inputmode={inputmode}
+      aria-describedby={ariaDescribedby}
+      max={max}
+      min={min}
+      onInput={e => console.log('input event value:', e.target.value)}
+      onBlur={e => console.log('blur event', e)}
+    />
+  );
+};
+
+export const Default = Template.bind({});
+Default.args = { ...defaultArgs };
+Default.argTypes = propStructure(numberInputDocs);
+
+export const Error = Template.bind({});
+Error.args = { ...defaultArgs, error: 'This is an error message' };
+
+export const Required = Template.bind({});
+Required.args = { ...defaultArgs, required: true };
+
+export const WithAnalytics = Template.bind({});
+WithAnalytics.args = { ...defaultArgs, 'enable-analytics': true };
+
+export const ValidRange = Template.bind({});
+ValidRange.args = {
+  ...defaultArgs,
+  min: 0,
+  max: 4,
+};
+
+export const AriaDescribedby = Template.bind({});
+AriaDescribedby.args = {
+  ...defaultArgs,
+  'error': 'This is also an error',
+  'aria-describedby': 'testing-id',
+};
```

Comment:
In order to test this with screen readers we may want to make sure the template renders with a `<div id="testing-id">This explains the error in detail</div>` or something like that.

Reply:
@bkjohnson so right now it outputs like:
```
<input id="inputField" type="number" aria-describedby="testing-id error-message" inputmode="numeric" name="my-input">
```
and that exists within the shadow DOM. Are you thinking that moving aria-describedby out into its own `div` with the prop setting the id and the error message setting the content above the input would be a better move? Should we then visually hide that element?

Reply:
I was imagining something like this for the story:

```jsx

return (
  <>
    <div id="testing-id">Some more detail on the error</div>
    <va-number-input error="You had an error" aria-describedby="testing-id" />
  </>
);

```

Because that's the way I understand `aria-describedby` to work. But it sounds like we'll be doing that with another ticket.